Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp

```diff
@@ -31,406 +31,143 @@
 namespace WebKit {
 using namespace WebCore;
 
-RefPtr<SecurityOrigin> NetworkResourceLoadParameters::parentOrigin() const
-{
-    if (frameAncestorOrigins.isEmpty())
-        return nullptr;
-    return frameAncestorOrigins.first();
-}
-
-void NetworkResourceLoadParameters::encode(IPC::Encoder& encoder) const
-{
-    encoder << identifier;
-    encoder << webPageProxyID;
-    encoder << webPageID;
-    encoder << webFrameID;
-    encoder << parentPID;
-    encoder << request;
-
-    encoder << static_cast<bool>(request.httpBody());
-    if (request.httpBody()) {
-        request.httpBody()->encode(encoder);
-
-        Vector<SandboxExtension::Handle> requestBodySandboxExtensions;
-        for (const FormDataElement& element : request.httpBody()->elements()) {
-            if (auto* fileData = std::get_if<FormDataElement::EncodedFileData>(&element.data)) {
-                const String& path = fileData->filename;
-                if (auto handle = SandboxExtension::createHandle(path, SandboxExtension::Type::ReadOnly))
-                    requestBodySandboxExtensions.append(WTFMove(*handle));
-            }
-        }
-        encoder << requestBodySandboxExtensions;
-    }
-
-    if (request.url().isLocalFile()) {
-        SandboxExtension::Handle requestSandboxExtension;
-#if HAVE(AUDIT_TOKEN)
-        if (networkProcessAuditToken) {
-            if (auto handle = SandboxExtension::createHandleForReadByAuditToken(request.url().fileSystemPath(), *networkProcessAuditToken))
-                requestSandboxExtension = WTFMove(*handle);
-        } else
+NetworkResourceLoadParameters::NetworkResourceLoadParameters(
+    NetworkLoadParameters&& networkLoadParameters
+    , WebCore::ResourceLoaderIdentifier identifier
+    , RefPtr<WebCore::FormData> httpBody
+    , std::optional<Vector<SandboxExtension::Handle>> sandboxExtensionIfHttpBody
```

Comment:
Should be an r-value to prevent unnecessary vector copies, same with frameAncestorOrigins.  Using r-values with other members also reduces ref churn, such as httpBody, sandboxExtensionIflocalFile (which should also have capitalization fixed), parentFrameURL, frameURL, documentURL, openerURL, and mainDocumentURL.  It wouldn't hurt to make all of them r-values.